### DISCUSSION TITLE
Trame connection closed after a few seconds

### DISCUSSION THREAD

**Discussion by Sichao25** [Kitware/trame#521] [Q&A] [answered]
Hi,

I am testing my application locally with Docker. However, I'm encountering an issue where the application briefly appears for one or two seconds before disappearing with a 'connection closed' message. The code I used is:
```
docker build -t trame-app .
docker run -it --rm -p 8080:80 trame-app
```
The log I got is:
```
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
                                                                                                                 [ OK ]
Starting server...
Starting the wslink launcher at
======== Running on http://0.0.0.0:9000 ========
(Press CTRL+C to quit)
```
I saw a similar discussion recommending changing the version of aiohttp, but unfortunately, this solution didn't work for me. I'm unsure what this issue implies and how to resolve it. Any suggestions or insights would be greatly appreciated.

**Comment by jourdain** [comment]
Any error in the session log?

**Reply by Sichao25** [reply]
Actually, I haven't figured out how to get session logs from Docker locally, as I am relatively new to this environment. When I deploy the same app on the server, there are no errors in the session logs. (While the same issue exists)
```
App running at:
 - Local:   http://0.0.0.0:9500/
 - Network: http://0.0.0.0:9500/

Note that for multi-users you need to use and configure a launcher.
```

**Reply by jourdain** [reply]
in docker the sessions logs are in `/deploy/server/logs/launcher/*.txt`

**Reply by Sichao25** [reply]
Thanks a lot. There are no errors in the Docker session logs as well. Logs are as follow:
```
App running at:
 - Local:   http://0.0.0.0:9500/
 - Network: http://0.0.0.0:9500/

Note that for multi-users you need to use and configure a launcher.
```

**Reply by jourdain** [reply]
It is hard to tell especially without error. ;-)

Do you have any error in the browser console?
Also what version do you have in your docker?

```
# inside docker
source /deploy/server/venv/bin/activate
pip list
```

**Reply by Sichao25** [reply]
Yes, I do see some errors in the browser console.


![browser_console](https://github.com/Kitware/trame/assets/54721834/f2cf8477-6d2c-4a88-a2db-338855dfef23)


 If you are talking about Trame versions, they are
```
trame                    2.5.2
trame-client             3.0.2
trame-components         2.3.2
trame-deckgl             2.0.3
trame-markdown           2.0.2
trame-matplotlib         2.0.3
trame-plotly             2.1.1
trame-rca                0.4.1
trame-router             2.2.0
trame-server             3.0.0
trame-simput             2.4.2
trame-vega               2.1.1
trame-vtk                2.8.8
trame-vuetify            2.5.0
```

**Reply by jourdain** [reply]
Thanks for sharing and indeed you do have version mismatch. 
Try to bump `trame>=3.6` in you dependency

**Reply by Sichao25** [reply]
Unfortunately, the issue still remains after setting trame>=3.6.0. The logs are the same. The new dependency list is:
```
trame                    3.6.0
trame-client             3.0.2
trame-components         2.3.2
trame-server             3.0.0
trame-vtk                2.8.8
trame-vuetify            2.5.0
```
I also explicitly set the client_type to "vue2" now, as it seems that it's no longer the default. Are there any other packages  I should be aware of to avoid a version mismatch, or should I modify my app to adapt "vue3"?

**Reply by jourdain** [reply]
1. How did you update the dependencies? Did you rebuild docker?
2. Did you clear your browser cache to make sure to pickup the new JS?
3. vue3 is not mandatory but if you can move to vue3, it will be better for the long term

**Reply by Sichao25** [reply]
Thanks for the follow-up.
1. I updated the setup/requirements.txt and rebuilt the docker.
2. Yes, I cleared the browser cache.
3. If I can't figure this out today, that will be my next step.

**Comment by jourdain** [comment]
Can you provide a full dependency list?

Technically I'm mainly interested in `wslink`, `msgpack`, `trame`, `trame-client`, `trame-server` .

**Reply by Sichao25** [reply]
I used `wslink 2.0.2`, `msgpack 1.0.8`, `trame 3.6.0`, `trame-client 3.0.2`, `trame-server 3.0.0`. The full list is as follows:
```
Package                  Version
------------------------ -----------
aiohttp                  3.9.5
aiosignal                1.3.1
anndata                  0.10.7
array_api_compat         1.6
async-timeout            4.0.3
attrs                    23.2.0
av                       12.0.0
certifi                  2024.2.2
charset-normalizer       3.3.2
contourpy                1.2.1
cycler                   0.12.1
exceptiongroup           1.2.1
filelock                 3.14.0
fonttools                4.51.0
frozenlist               1.4.1
fsspec                   2024.3.1
gpytorch                 1.11
h5py                     3.11.0
idna                     3.7
imageio                  2.34.1
imageio-ffmpeg           0.4.9
importlib_resources      6.4.0
jaxtyping                0.2.28
Jinja2                   3.1.4
joblib                   1.4.2
kiwisolver               1.4.5
linear-operator          0.5.2
MarkupSafe               2.1.5
matplotlib               3.8.4
more-itertools           10.2.0
mpmath                   1.3.0
msgpack                  1.0.8
multidict                6.0.5
natsort                  8.4.0
networkx                 3.2.1
numpy                    1.26.4
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        8.9.2.26
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.20.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.1.105
packaging                24.0
pandas                   2.2.2
pillow                   10.3.0
pip                      24.0
pkg_resources            0.0.0
platformdirs             4.2.1
pooch                    1.8.1
POT                      0.9.3
psutil                   5.9.8
pyacvd                   0.2.11
PyMCubes                 0.1.4
pymeshfix                0.16.3
pyparsing                3.1.2
python-dateutil          2.9.0.post0
pytz                     2024.1
pyvista                  0.40.0
requests                 2.31.0
scikit-learn             1.4.2
scipy                    1.13.0
scooby                   0.10.0
setuptools               44.0.0
six                      1.16.0
sympy                    1.12
threadpoolctl            3.5.0
torch                    2.3.0
trame                    3.6.0
trame-client             3.0.2
trame-components         2.3.2
trame-server             3.0.0
trame-vtk                2.8.8
trame-vuetify            2.5.0
triton                   2.3.0
typeguard                2.13.3
typing_extensions        4.11.0
tzdata                   2024.1
urllib3                  2.2.1
vtk                      9.3.0
wslink                   2.0.2
yarl                     1.9.4
zipp                     3.18.1
```

**Comment by jourdain** [comment]
Are you sure you are still having the same JS error in the dev tool console of your browser?

**Reply by Sichao25** [reply]
Yes

**Reply by jourdain** [reply]
And locally outside of docker, it is working fine?

**Reply by Sichao25** [reply]
Yes, it works fine locally within conda.

**Reply by jourdain** [reply]
can you share the content of your setup directory, along with the content of your DockerFile?

**Reply by Sichao25** [reply]
I am deploying the deployment branch of this [repo](https://github.com/Sichao25/spateo-viewer/tree/deployment). Thanks a lot for your time and support.

**Reply by jourdain** [reply]
Why do you have `timeout=0` in your app. You will starve the resources in docker with that arg.

**Reply by jourdain** [reply, marked as answer]
Based on your code you should have a XLib error in your session log when running with docker. 
In other words, the issue is that you are not using an offscreen version of VTK. 

Please find the right log to look into docker (`/deploy/server/logs/launcher/????.txt`) that will help you debug stuff down the road.

**Reply by Sichao25** [reply]
Fixed the issue by using an offscreen version of VTK. Sincere thanks for your time and help on this.
